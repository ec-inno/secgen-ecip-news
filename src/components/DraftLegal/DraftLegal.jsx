import React from 'react';
import { useTranslation } from 'react-i18next';

import File from '../File';

const DraftLegal = ({ file }) => {
  if (!file || Object.keys(file).length === 0) return '';

  const { t } = useTranslation();

  return (
    <>
      <h2 className="ecl-u-type-heading-2">{t('Draft legal act')}</h2>
      <File file={file} />
    </>
  );
};

export default DraftLegal;
