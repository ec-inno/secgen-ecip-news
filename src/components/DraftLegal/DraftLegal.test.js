import React from 'react';
import renderer from 'react-test-renderer';

import DraftLegal from './DraftLegal';

describe('DraftLegal', () => {
  it('no input: returns an empty string', () => {
    const tree = renderer.create(<DraftLegal />).toJSON();
    expect(tree).toBe('');
  });

  it('renders correctly provided an input', () => {
    const props = {
      file: {
        name: 'test',
        mimeType: 'text/plain',
        id: 16806,
        size: 10,
      },
    };

    const tree = renderer.create(<DraftLegal {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
