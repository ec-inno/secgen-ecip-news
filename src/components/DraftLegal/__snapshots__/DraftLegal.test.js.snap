// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DraftLegal renders correctly provided an input 1`] = `
Array [
  <h2
    className="ecl-u-type-heading-2"
  >
    Draft legal act
  </h2>,
  <div
    className="ecl-file"
    data-ecl-file={true}
  >
    <div
      className="ecl-file__container"
    >
      <svg
        aria-hidden="true"
        className="ecl-file__icon ecl-icon ecl-icon--2xl ecl-icon ecl-icon--m"
        focusable="false"
      >
        
        
        <use
          xlinkHref="file#general--copy"
        />
      </svg>
      <div
        className="ecl-file__info"
      >
        <div
          className="ecl-file__title"
        >
          test
        </div>
        <div
          className="ecl-file__meta"
        >
          (
          10 Bytes
           -
           
          TXT
          )
        </div>
      </div>
      <a
        className="ecl-link ecl-link--standalone"
        download={true}
        href="undefined/register/document/16806"
      >
        <span
          className="ecl-link__label"
        >
          Download
        </span>
         
        <svg
          aria-hidden="true"
          className="ecl-icon--fluid ecl-link__icon ecl-icon ecl-icon--m"
          focusable="false"
        >
          
          
          <use
            xlinkHref="file#ui--download"
          />
        </svg>
      </a>
    </div>
  </div>,
]
`;
